pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -227,6 +227,10 @@ func (c *collector) collectPodResUsed() {
 	podMetas := c.statesInformer.GetAllPods()
 	for _, meta := range podMetas {
 		pod := meta.Pod
+		// koordlet only supports standard OCI-runtime, such as runc.
+		if shouldIgnorePod(pod) {
```

Comment:
We may check the pod meta (e.g. pod runtimeclass, pod annotations) for the corresponding collector plugin, but not ignore all non-runc pods.

e.g. By default, we keep the collection for runc pods. For a kata-runtime pod, we collect its pod metric via a collector plugin of kata-runtime.

Reply:
I got your point. 
Although we ignore all non-runc pods here, users are still free to create their own plugins to collect Pods run by other runtimes.